integration/test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -93,6 +93,53 @@ func TestWriteLogsToCloudWatch(t *testing.T) {
 	}
 }
 
+// Validate https://github.com/aws/amazon-cloudwatch-agent/issues/447
+func TestRotatingLogsDoesNotSkipLines(t *testing.T) {
+	cfgFilePath := "resources/config_log_rotated.json"
+
+	// this uses the {instance_id} placeholder in the agent configuration,
+	// so we need to determine the host's instance ID for validation
+	ctx := context.Background()
+	c, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		// fail fast so we don't continue the test
+		t.Fatalf("Error occurred while creating SDK config: %v", err)
+	}
+
+	// TODO: this only works for EC2 based testing
+	client := imds.NewFromConfig(c)
+	metadata, err := client.GetInstanceIdentityDocument(ctx, &imds.GetInstanceIdentityDocumentInput{})
+	if err != nil {
+		t.Fatalf("Error occurred while retrieving EC2 instance ID: %v", err)
+	}
+	instanceId := metadata.InstanceID
+	log.Printf("Found instance id %s", instanceId)
+	logGroup := instanceId
+	logStream := instanceId + "Rotated"
```

Comment:
The reason I opted to use a different log stream name here is because as I was trying out the integration test, sometimes it would pick up log events from the other CWL integration test. Rather than introducing a sleep timer here to account for that, I just broke this out into it's own log group/stream.

Reply:
These tests should be running 1 at a time. What command did you use to start this test or did you run in the integration tests pipeline. If you ran in the pipeline and the tests were not running 1 at a time that is a bug.

Reply:
I can dig up one of the old pipeline test runs, but basically what happened was the call to GetLogEvents would pick up the very last log line from the previous test.

Reply:
```
aws_instance.integration-test (remote-exec):         	Error Trace:	cwl_util.go:127
aws_instance.integration-test (remote-exec):         	            				publish_logs_test.go:137
aws_instance.integration-test (remote-exec):         	Error:      	"[   Apr 25 18:21:43.997 - [bar] #99 This is a log line. 12345678901234567890123456789012345678901234567890]" should have 3 item(s), but has 5
```
Ignore the actual len assertion, I messed that up. But this test from https://github.com/SaxyPandaBear/amazon-cloudwatch-agent/runs/6162948889?check_suite_focus=true has `Apr 25 18:21:43.997 - [bar] #99 This is a log line.`, which is from the above test. So like I said, rather than introducing more sleeping (and potentially needing to bump up the test timeout), I just opted to direct these logs to a different stream.